src/pdfFiller/1040FieldMappings.ts

```diff
@@ -4,32 +4,27 @@ interface FieldMapping {
   [field: number]: (info: Information) => string | boolean | undefined
 }
 
-const depFields = (s: Information, depIdx: number): [string, string, string, boolean, boolean] | undefined => {
+const depField = (s: Information, pdfFieldIdx: number): string | boolean | undefined => {
   const deps: Dependent[] = s.taxPayer?.dependents ?? []
 
-  if (deps.length > depIdx) {
-    const dep: Dependent = deps[depIdx]
-    return [
-      dep.lastName, dep.ssid, dep.relationship, false, false
-    ]
-  }
-
-  return undefined
-}
+  // Based on the PDF row we are on, select correct dependent
+  const depIdx = Math.floor((pdfFieldIdx - 32) / 5)
 
-const depField = (s: Information, fieldIdx: number): string | boolean | undefined => {
-  const fs = depFields(s, Math.floor((fieldIdx - 32) / 5))
-  if (fs !== undefined) {
-    return fs[(fieldIdx - 32) % 5]
+  if (depIdx < deps.length) {
+    const dep = deps[depIdx]
+    // Based on the PDF column, select the correct field
+    const depFieldIdx = (pdfFieldIdx - 32) % 5
+    return [dep.lastName, dep.ssid, dep.relationship, false, false][depFieldIdx]
```

Comment:
Looks like only the last name is getting added for dependents. Looking at the 1040 form, it should be first name concatenated to last name  with a space in between

Reply:
done.